File: go-app/internal/models/grading_submission.go
package models

import (
	"database/sql"
	"time"

	"gorm.io/gorm"
)

type GradingSubmission struct {
	ID               uint           `gorm:"primaryKey" json:"id"`
	SubmissionName   string         `gorm:"column:submission_name" json:"submission_name"`
	SendDate         sql.NullTime   `gorm:"column:send_date" json:"send_date"`
	ReturnDate       sql.NullTime   `gorm:"column:return_date" json:"return_date"`
	Company          string         `json:"company"`
	Status           string         `json:"status"`
	SubmissionMethod string         `gorm:"column:submission_method" json:"submission_method"`
	Notes            sql.NullString `gorm:"type:text" json:"notes"`
	CostPerCard      float64        `gorm:"column:cost_per_card" json:"cost_per_card"`
	TaxRate          float64        `gorm:"column:tax_rate" json:"tax_rate"`
	SubmissionCost   float64        `gorm:"column:submission_cost" json:"submission_cost"`
	UpchargeTotal    float64        `gorm:"column:upcharge_total" json:"upcharge_total"`
	Items            []TrackedItem  `gorm:"foreignKey:GradingSubmissionID" json:"items"`
	CreatedAt        time.Time      `json:"created_at"`
	UpdatedAt        time.Time      `json:"updated_at"`
	DeletedAt        gorm.DeletedAt `gorm:"index" json:"-"`
}

func (s *GradingSubmission) GrandTotal() float64 {
	return s.SubmissionCost + s.UpchargeTotal
}

File: src/main/java/com/collectingwithzak/entity/GradingSubmission.java
package com.collectingwithzak.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "grading_submissions")
@Getter
@Setter
@SQLDelete(sql = "UPDATE grading_submissions SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
public class GradingSubmission extends BaseEntity {

    @Column(name = "submission_name")
    private String submissionName;

    private String company;

    private String status = "PREPPING";

    @Column(name = "submission_method")
    private String submissionMethod;

    @Column(name = "send_date")
    private LocalDate sendDate;

    @Column(name = "return_date")
    private LocalDate returnDate;

    private String notes;

    @Column(name = "cost_per_card", columnDefinition = "numeric(10,2)")
    private double costPerCard;

    @Column(name = "tax_rate", columnDefinition = "numeric(6,5)")
    private double taxRate;

    @Column(name = "submission_cost", columnDefinition = "numeric(10,2)")
    private double submissionCost;

    @Column(name = "upcharge_total", columnDefinition = "numeric(10,2)")
    private double upchargeTotal;

    @OneToMany(mappedBy = "gradingSubmission", fetch = FetchType.LAZY)
    private List<TrackedItem> items = new ArrayList<>();

    public double getGrandTotal() {
        return submissionCost + upchargeTotal;
    }
}
